include/swift/ABI/TaskLocal.h

```diff
@@ -32,62 +32,71 @@ class TaskGroup;
 
 class TaskLocal {
 public:
-  /// Type of the pointed at `next` task local item.
-  enum class NextLinkType : uintptr_t {
-    /// The storage pointer points at the next TaskLocal::Item in this task.
-    IsNext = 0b00,
-    /// The storage pointer points at a item stored by another AsyncTask.
+  class Storage;
+
+  /// Type of item in the task local item linked list.
+  enum class ItemKind : intptr_t {
+    /// Regular value item.
+    /// Has @c valueType and @c key .
+    /// Value is stored in the trailing storage.
+    /// @c next pointer points to another item owned by the same task as current item.
+    Value = -1,
+      
+    /// Item that marks end of sequence of items owned by the current task.
+    /// @c next pointer points to an item owned by another AsyncTask.
     ///
     /// Note that this may not necessarily be the same as the task's parent
     /// task -- we may point to a super-parent if we know / that the parent
     /// does not "contribute" any task local values. This is to speed up
-    /// lookups by skipping empty parent tasks during get(), and explained
-    /// in depth in `createParentLink`.
-    IsParent = 0b01,
+    /// lookups by skipping empty parent tasks during @c get() , and explained
+    /// in depth in @c createParentLink() .
+    ParentLink = 0,
+    
+    /// Stop-item that blocks further lookup.
+    /// Inserting stop-node allows to temporary disable all inserted task-local values in O(1),
+    /// while maintaining immutable linked list nature of the task-local values implementation.
+    Stop = 1,
```

Comment:
I'd like to use the `Barrier` terminology for this operation. Or `LookupBarrier` if we wanted to be longer.

Reply:
Done